File: TheBookLore/app/src/main/java/com/penguin/thebooklore/repository/ICollectionRepository.java
package com.penguin.thebooklore.repository;

import com.penguin.thebooklore.model.ArtObject;
import com.penguin.thebooklore.repository.network.model.CollectionResponse;

import java.util.List;

import io.reactivex.Single;

public interface ICollectionRepository {
    Single<CollectionResponse> getCollection();
}

File: TheBookLore/app/src/main/java/com/penguin/thebooklore/repository/ICollectionRepository.kt
package com.penguin.thebooklore.repository

import com.penguin.thebooklore.model.Artwork
import com.penguin.thebooklore.model.networkModel.Result

interface ICollectionRepository {
    suspend fun getCollection(type: String, ps: Int, p: Int): Result<List<Artwork>>
}
